woltka/biom.py

```diff
@@ -43,26 +43,28 @@ def profile_to_biom(profile, samples=None, tree=None, rankdic=None,
     biom.Table
         Converted BIOM table.
     """
-    features = sorted(allkeys(profile))
+    # features = sorted(allkeys(profile))
     samples = samples or sorted(profile)
-    data, metadata, names = [], [], []
-    for feature in features:
+    observations, data, metadata = [], [], []
 
+    for key in sorted(allkeys(profile)):
         # generate data
         row = []
         for sample in samples:
-            row.append(str(profile[sample][feature]) if feature in
-                       profile[sample] else 0)
+            row.append(profile[sample][key] if key in profile[sample] else 0)
         data.append(row)
-
-        # generate metadata
-        meta_ = {}
+        # generate metadata and observation
+        stratum, feature = key if isinstance(key, tuple) else (None, key)
+        obsrv, meta_ = feature, {}
         if namedic:
             name = namedic[feature] if feature in namedic else None
             if name_as_id:
-                names.append(name or feature)
+                obsrv = name or feature
             else:
                 meta_['Name'] = name or None
+        if stratum:
```

Comment:
Are the `if namedic` and `if stratum` branches mutually exclusive? It looks like it, since `obsrv` is overwritten by the second branch.

Would suggest making this clearer -- e.g. by using `elif`, or at least adding a comment.

Reply:
They are NOT exclusive. `namedic` will replace taxon IDs with descriptive names or append names to the table. `stratum` will prefix taxon ID/name with stratum label. For examples:

- original: `2697049`
- with `namedic`: `SARS-CoV-2`
- with `stratum`: `spike|2697049`
- with `namedic` and `stratum`: `spike|SARS-CoV-2`

Reply:
I added some inline comments to make the code more clear.